lib/ash/reactor/steps/create_step.ex

```diff
@@ -23,8 +23,22 @@ defmodule Ash.Reactor.CreateStep do
       |> maybe_set_kw(:authorize?, options[:authorize?])
 
     changeset =
-      options[:resource]
-      |> Changeset.for_create(options[:action], arguments[:input], changeset_options)
+      case arguments.initial do
+        changeset when is_struct(changeset, Changeset) ->
+          if Enum.any?(arguments.input) do
+            Changeset.force_change_attributes(changeset, arguments.input)
```

Comment:
I think this should do something like:

```elixir
changeset when is_nil(changeset.__validated_for_action__) ->
   Ash.Changeset.for_create(changeset, ....)

changeset when changeset.__validated_for_action__ == options[:action] ->
  # do your changing of attributes

changeset ->
  # this is probably an error? You gave me a changeset for a different action.
```

Or alternatively if its a changeset, pass it into `for_action` and let Ash.Changeset decide if that is a problem or not (getting a changeset already validated w/ a different action)